### ISSUE TITLE
There is a sync error after syncing supplier credit made from desktop to mobile

### ISSUE THREAD

**Issue by nishaSussol** [openmsupply/mobile#2801]
## Describe the bug
After creating a supplier credit to mobile store and syncing to mobile, sync looks messed up in mobile.

### To reproduce
1. Create a supplier credit for mobile store from desktop.
2. Try syncing with mobile.
3. See error. 

### Screenshots
![Screenshot_2020-04-29-16-39-32](https://user-images.githubusercontent.com/59193490/80594541-e99bd500-8a42-11ea-9123-d1c615094e66.png)

### Expected behaviour
There should be no sync errors,

### Version and device info
- App version: v5.0.0RC2
- Tablet model: Fire 5th gen

### Additional context
Found while testing #2756

**Comment by wlthomson** [comment]
@ArpitaSussol Phew! Took me hours to figure this out, but it comes back to a pretty simple bug:

```
// Auto-finalise all incoming consumer credits.
if (record.type === 'customer_credit' && !record.isFinalised) {
  funcs.push(() => {
    record.finalise();
  });
}
```

As `record` is a transaction, can find the `finalise()` method here:

```
finalise(database) {
  if (this.isFinalised) {
    throw new Error('Cannot finalise as transaction is already finalised');
  }

  // Prune all invoices except internal supplier invoices.
  if (!this.isInternalSupplierInvoice) {
    this.pruneRedundantBatchesAndItems(database);
  }
  ...
```

No `database` was being passed! This then caused no `database` to be passed to `pruneRedundantBatchesAndItems`, which then passes no `database` to `getTransactionBatches`, which then tries to do:

```
getTransactionBatches(database) {
  return database.objects('TransactionBatch').filtered('transaction.id == $0', this.id);
}
```

Which is throwing the error: `undefined is not an object (evaluating t.objects)`.

Took me ages to find this one because I just couldn't find where `.objects` was being called! Should have read the method signature more carefully!

**Comment by wlthomson** [comment]
Fixed by #2803.

**Comment by nishaSussol** [comment]
## TESTING
Tested in mobile v5.0.0RC3

### Setup
- To sync a customer credit to mobile, create a supplier credit for the mobile store on desktop.
- Customer credits can be viewed in customer invoices (should be finalised, so will have to toggle to "Past").

### Test cases
- [x] Customer credits are synced to mobile without any errors. **-PASS-**
- [x] Customer credits synced to mobile are auto-finalised. **-PASS-**
- [x] Customer credits synced to mobile cannot be edited. **-PASS-**
